File: CriminalIntent/app/src/main/java/com/bignerdranch/android/criminalintent/database/CrimeDataBase.java
package com.bignerdranch.android.criminalintent.database;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.bignerdranch.android.criminalintent.Crime;

@Database(entities = Crime.class, version = 1)
@TypeConverters(CrimeTypeConverters.class)
// abstract: 抽象;    -- 👋🐘;
public abstract class CrimeDataBase extends RoomDatabase {
}

File: CriminalIntent2/app/src/main/java/com/bignerdranch/android/criminalintent/database/CrimeDatabase.kt
package com.bignerdranch.android.criminalintent.database

import android.os.Parcel
import android.os.Parcelable
import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.bignerdranch.android.criminalintent.Crime

@Database(entities = [ Crime::class ], version = 1, exportSchema = false)
@TypeConverters(CrimeTypeConverters::class)
abstract class CrimeDatabase() : RoomDatabase() {
    abstract fun crimeDao(): CrimeDao;
}
